src/github_repo_request_local.py

```diff
@@ -101,27 +220,74 @@ def list_test_files(directory, excluded_extensions):
                 capture_output=True,
             )
             logger.info(f"Successfully cloned the repo: {repo_name}")
+
+            # Fetch the last commit hash and store it in the DataFrame
+            last_commit_hash = get_last_commit_hash(clone_dir)
+            df.at[index, "last_commit_hash"] = last_commit_hash
```

Comment:
Taking into account several of my earlier comments, we'd need to revise this code to only store the last commit hash if it was returned by the function that obtains it i.e. if the value is not `None`.

Reply:
The column `last_commit_hash` is being initialised with `None`. I have added a condition to skip storing the value if the function returns `None`.